docs/learn/encyclopedia/data-format/xdr.mdx

```diff
@@ -16,6 +16,27 @@ Since XDR is a binary format and not as widely known as simpler formats like JSO
 
 In addition, the Horizon API server generally exposes the most important parts of the XDR data in JSON, so they are easier to parse if you are not using an SDK. The XDR data is still included (encoded as a base64 string) inside the JSON in case you need direct access to it.
 
+
+## JSON and XDR Conversion Schema
+
+:::info
+
+SDF has published libraries which support the canonical and round-trippable conversion from XDR to JSON, and from JSON to XDR, without loss of information.
+
+:::
+
+The canonical JSON and XDR Schema is defined by the [stellar_xdr crate](https://docs.rs/stellar-xdr/21.1.0/stellar_xdr/index.html) and by the [@stellar/stellar-xdr-json-web npm package](https://www.npmjs.com/package/@stellar/stellar-xdr-json-web).
+
+This conversion is visible in the Stellar CLI, Laboratory View XDR, and Hubble events table. The schema provides a round-trippable means for converting any Stellar XDR type to JSON and converting that JSON back to the identical XDR.
+
+### Key Characteristics of the JSON and XDR Conversion Schema
+
+- Round-Trippable: The JSON format allows for converting from XDR to JSON and back to XDR without loss of information.
+
+- Self-Describing: The JSON format describes the internals of the type but does not identify the type that is encoded. This is similar to XDR, which also does not identify the encoded type.
+
+- Single Valid Representation: For any given XDR, there is a single valid JSON representation, and vice-versa. This ensures reproducibility and consistency.
+
 ## .X files
 
-Data structures in XDR are specified in an interface definition file (IDL). The IDL files used for the Stellar Network are available on [GitHub](https://github.com/stellar/stellar-core/tree/master/src/protocol-curr/xdr).
+Data structures in XDR are specified in an interface definition file (IDL). The IDL files used for the Stellar Network are available on [GitHub](https://github.com/stellar/stellar-xdr).
```

Comment:
We should move the section about the .X files to the top. The definitions and source of truth for the XDR should be the first thing on this page imo, with ways to access that information following after it.

Reply:
Really great idea!

Reply:
I took a stab at fixing this in [74fe590](https://github.com/stellar/stellar-docs/pull/820/commits/74fe590f650ecefd178a86affcf5c39854517e20), does that fit what you had in mind?

Reply:
Looks good thanks @ElliotFriend